Mathlib/Analysis/Complex/IntegerCompl.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.Complex.UpperHalfPlane.Basic
+
+/-!
+# Integer Complement
+
+We define the complement of the integers in the complex plane and give some basic lemmas about it.
+We also show that the upper half plane embeds into the integer complement.
+
+-/
+
+open UpperHalfPlane
+
+/--The complement of the integers in `ℂ`. -/
+def ComplexIntegerComplement := {z : ℂ // ¬ ∃ (n : ℤ), z = ↑n}
+
+namespace ComplexIntegerComplement
+
+local notation "ℂ_ℤ " => ComplexIntegerComplement
+
+noncomputable instance : UniformSpace ℂ_ℤ  :=  instUniformSpaceSubtype
+
+theorem Complex.closedEmbedding_coe_complex : ClosedEmbedding ((↑) : ℤ → ℂ) := by
```

Comment:
Please move to `Mathlib.Analysis.Complex.Basic` as `Complex.closedEmbedding_intCast`